docs/source/user-guide/supported_models.rst

```diff
@@ -0,0 +1,65 @@
+.. _supported_models:
+
+====================
+Supported Models
+====================
+
+Similar to scikit-learn API, a model will be compatible with deepchecks' checks if it implements some minimal functionality.
+Without a compatible model, deepchecks can't run all checks that require predictions, for example all :docs:`performance checks </api/checks/performance>`.
+
+Below are the minimal requirements for each task type:
+
+Required API
+=============
+
+Regression
+----------
+
+All that is necessary for a regression model is the ``predict`` function.
+The predict function should expect an `array-like` or `sparse matrix` of shape ``(n_samples, n_features)``
+and is expected to return an `ndarray` of shape ``(n_samples,)``, a vector containing the predicted value for each sample.
+
+
+Example of a simple model:
+
+>>> class simple_regression_model:
+...     def predict(X: pd.DataFrame) -> pd.Series:
+...         ...
+
+Classification
+--------------
+
+For classification models, we require also the ``predict_proba`` function.
```

Comment:
Do we require "also" or only?
I'm thinking if there is a way to put these two together (seems as if it's not that convenient to read... cause if I have a classification model, then I jump here, but then it says "we require also", so it doesn't make much sense...)